sql/src/main/java/io/crate/planner/Planner.java

```diff
@@ -258,12 +272,13 @@ public Planner(ClusterService clusterService, Functions functions, TableStats ta
      */
     public Plan plan(Analysis analysis, UUID jobId, int softLimit, int fetchSize) {
         AnalyzedStatement analyzedStatement = analysis.analyzedStatement();
+        RoutingProvider routingProvider = new RoutingProvider(Randomness.get().nextInt(), awarenessAttributes);
```

Comment:
I wonder whether the production scope constructor should have a constructor like that. You could potentially pass a constant seed.